main.py

```diff
@@ -14,21 +13,21 @@
 ap.add_argument("-b", "--build-number", required=False, help="Force a particular build number")
 ap.add_argument("-gl", "--get-license", required=False,
                 help="If set to true, it will ignore all other procedure and just create a license file")
-ap.add_argument("-f", "--config-file", required=False, help="Path to the config file")
-ap.add_argument("-l", "--log-level", required=False, help="Log level to run this program")
+ap.add_argument("-f", "--config-file", required=False, help="Path to the config file", default=None)
+ap.add_argument("-l", "--log-level", type=str, required=False, help="Log level to run this program", default="INFO")
 args = vars(ap.parse_args())
 
 
 def main():
-    setup_logger(args.get("log_level", "INFO"))
-    config_file = args.get("config_file", None)
+    setup_logger(args["log_level"])
 
-    config = Config(args, config_file)
+    config = Config(args, args["config_file"])
```

Comment:
Since you are already passing args to the config constructor, wouldn't it be better to just grab the config file path in the init method? One less parameter for the class constructor.

Reply:
fixed though it looks weird now. entire config needs rewrite probably

Reply:
actually i reverted that because it messes up with the order of handle_config_file and handle_args. this needs deeper rewrite than just removing parameter